### PR TITLE
meta: refine sequence interface for br to use

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve?

Problem Summary:  
Sequence object has two kv pair storing the current value and cycle round times. This two value is important for sequence to compute the next value. So it's better to supply
a interface out for back up this two values.

### What is changed and how it works?

What's Changed: refine the sequecne allocator interface int `autoid.go` 

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 
- refine the sequence interface for br to use, including supply the way to get sequence back up base and cycle round times.

### CODE REVIEW

**Comment by overvenus** [comment]
PTAL @3pointer

File: meta/autoid/autoid.go [hunk 1]
```diff
@@ -941,3 +955,67 @@ func (l *AutoRandomIDLayout) IncrementalBitsCapacity() uint64 {
 func (l *AutoRandomIDLayout) IncrementalMask() int64 {
 	return (1 << l.IncrementalBits) - 1
 }
+
+// GetSequenceBackUpBase implements autoid.Allocator GetSequenceBackUpBase interface.
+// This function is used for backup with sequence setval.
+func (alloc *allocator) GetSequenceBackUpBase(tableID int64) (int64, error) {
```
Comment (tangenta) [inline meta/autoid/autoid.go:961]: How is it different from `allocator.NextGlobalAutoID`?

Comment (AilinKid) [reply meta/autoid/autoid.go:961]: actually, it's quite same.
NextGlobalAutoID will return next applicable id, the upper use it directly, or the br rebase auto_increment with (id-1).

For sequence rebase/backup, the only way is via setval. the base parameter in setval() means the (id) is already been used, and next value will skip it and find next value.

Actually they(BR) can use NextGloalAutoID directly, and minus one for increment > 0 or plus one for increment < 0 cases cause NextGlobalAutoID will always return (id+1) regradless sequence feature.

**Comment by AilinKid** [comment]
since BR has implemented the code with another way, this pr is not necessary.

**Pull request closed without merging**